.github/workflows/label-pull-requests.yml

```diff
@@ -0,0 +1,153 @@
+                            label => !(label === prNeedsReviewLabel || label === prWaitingForAuthorLabel)
+                        );
+
+                        if (!newLabels.includes(prMaintainerApprovedButAwaitingCiLabel))
+                            newLabels.push(prMaintainerApprovedButAwaitingCiLabel);
+                    } else {
+                        if (!newLabels.includes(prCommunityApprovedLabel))
+                            newLabels.push(prCommunityApprovedLabel);
+                    }
+                } else if (!isBlocked) {
+                    newLabels = newLabels.filter(
+                        label =>
+                            !(
+                                label === prCommunityApprovedLabel ||
+                                label === prMaintainerApprovedButAwaitingCiLabel
+                            )
+                    );
+
+                    if (review.user.login === pull_request.user.login) newLabels.push(prNeedsReviewLabel);
+                    else newLabels.push(prWaitingForAuthorLabel);
+                }
+
+                return newLabels;
+            }
+
+            async function labelsForPullRequestEffectivelyClosed(currentLabels) {
+                return removeExistingPrLabels(currentLabels, false);
+            }
+
+            const eventHandlers = {
+                opened: labelsForGenericPullRequestChange,
+                reopened: labelsForGenericPullRequestChange,
+                submitted: labelsForPullRequestReviewSubmitted,
+                dismissed: labelsForGenericPullRequestChange,
+                converted_to_draft: labelsForPullRequestEffectivelyClosed,
+                ready_for_review: labelsForGenericPullRequestChange,
+                synchronize: labelsForGenericPullRequestChange, // synchronize is triggered when the branch is changed
+                edited: labelsForGenericPullRequestChange,
+                review_requested: labelsForGenericPullRequestChange,
+                closed: labelsForPullRequestEffectivelyClosed,
+            };
+
+            const eventName = context.payload.action;
+            const handlerForCurrentEvent = eventHandlers[eventName];
+
+            function apiErrorHandler(error) {
+                console.log(
+                    "::warning::Encountered error during event handling, not updating labels. Error:",
+                    error
+                );
+            }
+
+            async function updateLabels(currentLabelsAsObjects) {
+                const currentLabels = [];
+                currentLabelsAsObjects.forEach(labelObject => currentLabels.push(labelObject.name));
+                console.log("currentLabels", currentLabels);
+
+                const isEffectivelyClosed =
+                    context.payload.pull_request.draft ||
+                    context.payload.pull_request.state.toLowerCase() === "closed";
+
+                const newLabels = await (!isEffectivelyClosed
```

Comment:
nit pick: Avoid needlessly using boolean negation, we can write the same statement without by just swapping the ternary arguments. It's arguably easier to read code "straight" instead of having to invert the logic (especially for a ternary).